src/ophys_etl/transforms/roi_transforms.py

```diff
@@ -127,3 +129,142 @@ def crop_roi_mask(roi_mask: coo_matrix) -> coo_matrix:
     cropped_mask = roi_mask.tocsr()[min_row:max_row, min_col:max_col]
 
     return cropped_mask.tocoo()
+
+
+def coo_rois_to_old(coo_masks: List[coo_matrix],
+                    max_correction_vals: motion_border,
+                    movie_shape: Tuple[int, int]) -> List[Dict[str, Any]]:
+    """
+    Converts coo formatted ROIs to old format from previous Ophys Segmentation
+    implementation. This is a required transformation as the next workflow
+    step, Extract Traces, expects ROIs in this format and to reduce work load
+    this will not be changed.
+    Parameters
+    ----------
+    coo_masks: List[coo_matrix]
+        A list of scipy coo_matrices representing ROI masks, each element of
+        list is a unique ROI.
+    max_correction_vals: motion_border
+        The max motion correction values identified in the motion correction
+        step of ophys segmentation pipeline
+    movie_shape: Tuple[int, int]
+        The frame shape of the movie from which ROIs were extracted in order
+        of: (height, width).
+
+    Returns
+    -------
+    List[Dict[str, Any]]
+        A list of dictionaries representing the ROIs in old segmentation
+        format, the data contained inside each one is as follows:
+        {
+            id: int
+            x: int
+            y: int
+            width: int
+            height: int
+            valid_roi: Bool
+            mask_matrix: List[List[Bool]]
+            max_correction_up: int
+            max_correction_down: int
+            max_correction_left: int
+            max_correction_right: int
+            mask_image_plane: int
+            exclusion_labels: List[int] codes are defined in 2019 Ophys Docs
+        }
+        For details about specific values see design document for 2020 Ophys
+        Segmentation Refactor and Update
+        """
+    old_rois = []
+    for temp_id, coo_mask in enumerate(coo_masks):
+        old_roi = _coo_mask_to_old_format(coo_mask)
+        old_roi['id'] = temp_id  # popped off writing to LIMs
+        old_roi['cell_specimen_id'] = temp_id  # updated post nway cellmatching
+        old_roi['valid_roi'] = True
+        old_roi['max_correction_up'] = int(max_correction_vals.up)
+        old_roi['max_correction_down'] = int(max_correction_vals.down)
+        old_roi['max_correction_right'] = int(max_correction_vals.right)
+        old_roi['max_correction_left'] = int(max_correction_vals.left)
+        old_roi['mask_image_plane'] = 0
+        old_roi['exclusion_labels'] = []
+        _check_motion_exclusion(old_roi, movie_shape)
```

Comment:
I don't like mixing functions that do inplace modification with explicit setting, because it's not clear. I had to read the function to understand that we weren't saying all ROIs were valid.

I recommend to have _check_motion_exclusion return the appropriate values instead of doing in place modification. Name the function more generally so that it can be extended to check for all types of "exclusion" if more are added (can call the original function logic).

Something like:

```
old_roi["valid_roi"], old_roi["exclusion_labels"] = _check_validity(old_roi)

def _check_validity(old_roi: Dict, ...) -> Tuple[Bool, List[Int]]: ...
```

Reply:
Good suggestion, I can see that as being confusing. I changed the logic of the function to return that Tuple and also made the function name more generic so we can add on to it later!